File: ClientApp/src/app/shared/Models/service.ts
import { JsonApiModelConfig, JsonApiModel, Attribute, HasMany, BelongsTo } from 'angular2-jsonapi';
import { BaseTitleObject } from './_baseTitleObject';
import { Servicetype } from './servicetype';
import { Company } from './company';
import { Unit } from './unit';

@JsonApiModelConfig({ type: 'services' })
export class Service extends BaseTitleObject {

  @BelongsTo()  company: Company;
  @BelongsTo()  unit: Unit;
  @BelongsTo()  servicetype: Servicetype;

  @Attribute()  comment: string;
}

File: Models/DB/Service.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using JsonApiDotNetCore.Models;

namespace SURV.Models.DB {
    public class Service : BaseTitleObject {
        [ForeignKey (nameof (Servicetype))] public int? ServicetypeId { get; set; }

        [HasOne, Required] public virtual Servicetype Servicetype { get; set; }

        [ForeignKey (nameof (Unit))] public int? UnitId { get; set; }

        [HasOne, Required] public virtual Unit Unit { get; set; }

        [ForeignKey (nameof (Company))] public int? CompanyId { get; set; }

        [HasOne, Required] public virtual Company Company { get; set; }

        [Attr ("comment")] public string Comment { get; set; }
    }
}